File: src/slang/parse/PostfixOpParselet.java
package slang.parse;

import slang.lex.Token;

public class PostfixOpParselet implements InfixParselet {
    @Override
    public Expr parse(Parser parser, Expr left, Token token) {
        return new Expr.Postfix(left, token);
    }

    @Override
    public int getPrecedence() {
        return Precedence.POSTFIX;
    }
}

File: src/slang/parse/PostfixOpParselet.scala
package slang.parse

import slang.lex.Token

class PostfixOpParselet extends InfixParselet {
    override def parse(parser: Parser, left: Expr, token: Token): Expr = Expr.Postfix(left, token) 
    override def getPrecedence: Int = Precedence.POSTFIX
}
